src/NHibernate.Test/NHSpecificTest/GH1712/Fixture.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Linq;
+using NUnit.Framework;
+
+namespace NHibernate.Test.NHSpecificTest.GH1712
+{
+	[TestFixture]
+	public class Fixture : BugTestCase
+	{
+		protected override void OnSetUp()
+		{
+			using (var session = OpenSession())
+			using (var tx = session.BeginTransaction())
+			{
+				session.Save("GenericEntityWithGuid", new GenericEntity<Guid>(),
+					Guid.Parse("093D2C0D-C1A4-42CB-95FC-1039CD0C00B6"));
+				session.Save("GenericEntityWithLong", new GenericEntity<long>(), 42L);
+				session.Save("GenericEntityWithString", new GenericEntity<string>(), "Bob l'éponge");
+				tx.Commit();
+			}
+		}
+
+		protected override void OnTearDown()
+		{
+			base.OnTearDown();
+			using (var session = OpenSession())
+			using (var tx = session.BeginTransaction())
+			{
+				session.Delete("from GenericEntityWithGuid");
+				session.Delete("from GenericEntityWithLong");
+				session.Delete("from GenericEntityWithString");
+				tx.Commit();
+			}
+		}
+
+		[Test]
+		public void PerformingAQueryUsingIEquatableShouldNotThrowEquals()
+		{
+			using (var session = Sfi.OpenSession())
+			{
+				GenericEntity<TId> Query<TId>(string name, TId id) where TId : IEquatable<TId>
+					=> (from e in session.Query<GenericEntity<TId>>(name)
+						where e.Id.Equals(id)
+						select e).FirstOrDefault();
```

Comment:
The async generator currently fails handling the local function conversion. Better use a regular class function.

@maca88, it sometime generates nothing, other times generates invalid code: the async counterpart is found in `NHibernate.Linq` namespace which is not used in the sync file, and the async file is lacking the `using`.

It is not lacking when a regular class function is used. The trouble occurs with 0.11.1 version too.

Reply:
This is really odd, I've run the generator 5 times (version `0.8.2.4`) on my computer and the local function was always converted correctly and the `using` statement was always there. Also, the `WhereEquatableEqualAsync` was never generated in those 5 times. The only thing that I found is that the namespace `NHibernate.Linq` was never added for which I filled an [issue](https://github.com/maca88/AsyncGenerator/issues/101).

I also tried with version `0.11.1` and I got the same results as with `0.8.2.4`.

Reply:
Oh my bad, I don't know why I thought that the `using` was referred to the `using` statement used for the session, but after rereading your comment it is clear that it is referred to the `using` for the namespace. 
With the version `0.8.2.5` the missing `using` statement will now be generated.

Reply:
@maca88 can you push missing tags to the repo?

Reply:
I've added the missing tags and pushed them.